### PR TITLE
lazy load panels on scroll

### PR DESCRIPTION
**Pull request by sjcobb**
To help with performance this PR lazy loads panel content on scroll using react-intersection-observer

### CODE REVIEW

File: ui/dashboards/src/components/Panel/Panel.tsx [hunk 1]
```diff
@@ -37,11 +38,18 @@ export function Panel(props: PanelProps) {
     return { width, height };
   }, [width, height]);
 
+  const { ref, inView } = useInView({
+    threshold: 0,
+    initialInView: true,
+    triggerOnce: true,
```
Comment (LukeTillman) [inline ui/dashboards/src/components/Panel/Panel.tsx:44]: Ahh nice. Does this option mean that once it's in view and rendered, it doesn't unmount (and then get re-rendered later)?

**Review by LukeTillman — APPROVED** [review]
💥 This is gonna be great for dashboards with lots of charts, nice work.

**Comment by eunicorn** [comment]
nice, the api is so simple!

File: ui/dashboards/src/components/Panel/Panel.tsx (hunk 1 above)
Comment (sjcobb) [reply ui/dashboards/src/components/Panel/Panel.tsx:44]: Yep that's right! Don't want to rerun a bunch of queries if we don't have to

**Pull request merged**